statescript/executor.go

```diff
@@ -152,10 +154,51 @@ func (l Launcher) getTimeout() time.Duration {
 	return t
 }
 
+// returns true if the script has logged less than 10KB
+func scriptHasLogStorageLeft(name string, bytes int) bool {
+
+	db := store.NewDBStore("/data/mender")
+	if db == nil {
+		log.Error("Failed to open the database")
+	}
+	if bytes > 10*1024*8 { // more than 10KB will never be logged
+		return false
+	}
+
+	bused, err := db.ReadAll(name)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = db.WriteAll(name, []byte{byte(bytes)})
+			if err != nil {
+				log.Errorf("Failed to write storage data for %s to the database", name)
+			}
+			return true
+		}
+		log.Errorf("Failed to read the storage data for %s from the database", name)
+	}
+
+	totBytes := int(bused[0]) + bytes
+	if totBytes <= 10*1024*8 {
+		err = db.WriteAll(name, []byte{byte(totBytes)})
+		if err != nil {
+			log.Errorf("Failed to write storage data for %s to the database", name)
+		}
+		return true
+	}
+
+	return false
+}
+
 func execute(name string, timeout time.Duration) int {
 
 	cmd := exec.Command(name)
 
+	var stderr io.ReadCloser
+	var err error
+	if !strings.Contains(name, "idle") && !strings.Contains(name, "Sync") { // do not log from these states
```

Comment:
Shouldn't we have `Idle` here (capital letter)? What is more, I don't think we need this conditional path. It's not wrong to keep and maybe we should to be on a safe side, but the logger is not enabled if we are inside Idle or Sync state anyway, so we won't be able to log anything.